### PR TITLE
Witness fix

### PR DESCRIPTION
**Pull request by LukaszRozmej**

### CODE REVIEW

File: src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs [hunk 1]
```diff
@@ -695,5 +697,41 @@ private void PersistOnShutdown()
         }
 
         #endregion
+        
+        public IReadOnlyTrieStore AsReadOnly(IKeyValueStore readOnlyStore)
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs:701]: this is strange -> you modify the interface from a KeyValueStore to TrieStore and call it AsReadOnly
has to be changed

File: src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs [hunk 2]
```diff
@@ -695,5 +697,41 @@ private void PersistOnShutdown()
...
+        {
+            return new ReadOnlyTrieStore(this, readOnlyStore);
+        }
+        
+        private class ReadOnlyTrieStore : IReadOnlyTrieStore
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs:706]: why internal class, keep separately, in a separate file
this class is used externally

File: src/Nethermind/Nethermind.Trie/Pruning/IReadOnlyTrieStore.cs [hunk 3]
```diff
@@ -17,11 +17,8 @@
 
 namespace Nethermind.Trie.Pruning
 {
-    public static class TrieNodeResolverExtensions
+    public interface IReadOnlyTrieStore : ITrieStore
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Trie/Pruning/IReadOnlyTrieStore.cs:20]: bring back the extension method, can stay with an interface but do not move everything to a single class

File: src/Nethermind/Nethermind.AuRa.Test/Contract/AuRaContractGasLimitOverrideTests.cs [hunk 4]
```diff
@@ -97,7 +97,7 @@ protected override BlockProcessor CreateBlockProcessor()
                 var gasLimitContract = new BlockGasLimitContract(new AbiEncoder(), blockGasLimitContractTransition.Value, blockGasLimitContractTransition.Key,
                     new ReadOnlyTxProcessingEnv(
                         DbProvider,
-                        new TrieStore(DbProvider.StateDb, LimboLogs.Instance),
+                        new TrieStore(DbProvider.StateDb, LimboLogs.Instance).AsReadOnly(DbProvider.StateDb),
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.AuRa.Test/Contract/AuRaContractGasLimitOverrideTests.cs:100]: do we need to pass the DB if it is the same

File: src/Nethermind/Nethermind.AuRa.Test/Transactions/TxCertifierFilterTests.cs [hunk 5]
```diff
@@ -141,8 +141,8 @@ protected override BlockProcessor CreateBlockProcessor()
                 AbiEncoder abiEncoder = new AbiEncoder();
                 ReadOnlyTransactionProcessorSource = new(
                     DbProvider,
-                    new ReadOnlyTrieStore(new TrieStore(DbProvider.StateDb, LimboLogs.Instance)),
-                    BlockTree, SpecProvider,
+                    new TrieStore(DbProvider.StateDb, LimboLogs.Instance).AsReadOnly(DbProvider.StateDb),
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.AuRa.Test/Transactions/TxCertifierFilterTests.cs:144]: same DB used

File: src/Nethermind/Nethermind.AuRa.Test/Transactions/TxPermissionFilterTest.cs [hunk 6]
```diff
@@ -257,7 +257,7 @@ protected override BlockProcessor CreateBlockProcessor()
                 TransactionPermissionContractVersions =
                     new LruCache<Keccak, UInt256>(PermissionBasedTxFilter.Cache.MaxCacheSize, nameof(TransactionPermissionContract));
 
-                var trieStore = new ReadOnlyTrieStore(new TrieStore(DbProvider.StateDb, LimboLogs.Instance));
+                var trieStore = new TrieStore(DbProvider.StateDb, LimboLogs.Instance).AsReadOnly(DbProvider.StateDb);
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.AuRa.Test/Transactions/TxPermissionFilterTest.cs:260]: AsReadOnly() would be better here

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.AuRa.Test/Transactions/TxPermissionFilterTest.cs:260]: Would only work on TrieStore or we need to put it into the interface.

**Review by tkstanczak — APPROVED** [review]

**Pull request merged**